File: backend/src/main/java/net/ddns/lexdev/systempro_api/dto/SupplierRequestDto.java
package net.ddns.lexdev.systempro_api.dto;

import java.math.BigDecimal;
import java.util.List;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;

public record SupplierRequestDto(

    @NotNull
    @Valid
    PersonRequestDto person,

    @Valid
    IndividualPersonRequestDto individual,

    @Valid
    LegalEntityRequestDto legalEntity,

    @Valid
    List<PersonContactRequestDto> contacts,

    @Valid
    List<PersonAddressRequestDto> addresses,

    String condicaoPagamentoPadrao,

    Integer prazoEntregaDias,

    BigDecimal valorMinimoPedido,

    String categoria,

    String observacoesComerciais,

    BankDetailsDto bankDetails,

    List<SupplierContactDto> contatos,

    List<SupplierDocumentDto> documentos,

    Boolean active

) {}

File: frontend/src/app/features/models/supplier-request-dto.ts
import { BankDetailsDto } from "./bank-details-dto";
import { PersonAddressResponseDto } from "./person-address-response-dto";
import { PersonContactResponseDto } from "./person-contact-response-dto";
import { SupplierContactDto } from "./supplier-contact-dto";
import { SupplierDocumentDto } from "./supplier-document-dto";

export interface SupplierRequestDto {
  person: {
    tipoPessoa: string;
    name: string;
    cpfCnpj: string;
    individualPerson?: { rg: string };
    legalEntity?: { nomeFantasia?: string; inscricaoEstadual: string };
    contacts?: Omit<PersonContactResponseDto, 'id'>[];
    addresses?: Omit<PersonAddressResponseDto, 'id'>[];
  };
  condicaoPagamentoPadrao: string;
  prazoEntregaDias: number;
  valorMinimoPedido: number;
  categoria: string;
  observacoesComerciais: string;
  bankDetails?: BankDetailsDto;
  contatos?: SupplierContactDto[];
  documentos?: SupplierDocumentDto[];
  active?: boolean;
}
